### PR TITLE
Let the origin server header flow through http.proxy service

### PR DESCRIPTION
**Pull request by jitsni**

### CODE REVIEW

**Comment by dpwspoon** [comment]
LGTM.

File: service/http.proxy/src/test/java/org/kaazing/gateway/service/http/proxy/SecureOriginServer.java [hunk 1]
```diff
@@ -58,11 +58,14 @@ public void run() {
...
-                } catch (Exception ioe) {
+                } catch (IOException ioe) {
                     ioe.printStackTrace();
+                    if (this.ioe != null) {
+                        this.ioe = ioe;
+                    }
                 }
             } catch (IOException ioe) {
-                this.ioe = ioe;
+                // no-op since socket.accept() may throw IOE
```
Comment (brennangaunce) [inline service/http.proxy/src/test/java/org/kaazing/gateway/service/http/proxy/SecureOriginServer.java]: Please log this exception at the trace level.  No exception should ever go uncaught.

File: transport/http/src/main/java/org/kaazing/gateway/transport/http/HttpAcceptProcessor.java [hunk 2]
```diff
@@ -361,7 +361,7 @@ public static void setServerHeader(DefaultHttpSession httpSession, HttpResponseM
         if (httpSession != null) {
             ResourceAddress address = httpSession.getLocalAddress();
             boolean serverHeaderEnabled = address.getOption(HttpResourceAddress.SERVER_HEADER_ENABLED);
-            if (serverHeaderEnabled) {
+            if (serverHeaderEnabled && !httpResponse.hasHeader("Server")) {
```
Comment (brennangaunce) [inline transport/http/src/main/java/org/kaazing/gateway/transport/http/HttpAcceptProcessor.java]: The code above appears to set the header if server header is enabled, but do not overwrite.  In addition, when the server header is disabled the Server header is passed through.

Should we have an enhancement to *remove* the passthrough Server header?  (Valid options might be: Enabled, Disabled, Remove, Overwrite)

File: service/http.proxy/src/test/java/org/kaazing/gateway/service/http/proxy/SecureOriginServer.java (hunk 1 above)
Comment (jitsni) [inline service/http.proxy/src/test/java/org/kaazing/gateway/service/http/proxy/SecureOriginServer.java]: accept() will throw an exception when the socket is closed. So it is an expected exception.

File: transport/http/src/main/java/org/kaazing/gateway/transport/http/HttpAcceptProcessor.java (hunk 2 above)
Comment (jitsni) [inline transport/http/src/main/java/org/kaazing/gateway/transport/http/HttpAcceptProcessor.java]: Good point. I was thinking about it too. I thought there would be ways to configure at the origin server level. Moreover, I think more generic solution to configure any header.

File: service/http.proxy/src/test/java/org/kaazing/gateway/service/http/proxy/SecureOriginServer.java (hunk 1 above)
Comment (brennangaunce) [inline service/http.proxy/src/test/java/org/kaazing/gateway/service/http/proxy/SecureOriginServer.java]: Please log it at TRACE level.  No exception should ever go uncaught.

**Pull request merged**